core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -125,10 +126,46 @@ public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionD
     }
 
     public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> acknowledge(
+            String memberId,
             String groupId,
             Map<TopicIdPartition, ShareAcknowledgeRequestData.AcknowledgePartition> acknowledgeTopics
     ) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        log.debug("Acknowledge request for topicIdPartitions: {} with groupId: {}",
+                acknowledgeTopics.keySet(), groupId);
+        Map<TopicIdPartition, CompletableFuture<Errors>> futures = new HashMap<>();
+        synchronized (this) {
```

Comment:
This is currently synchronized across the entire broker, which seems excessive.

Reply:
Thanks for the review. I have made the changes in the latest commit